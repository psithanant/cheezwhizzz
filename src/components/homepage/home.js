import React, {Component} from 'react';
import { Grid, Col } from 'react-bootstrap';
import Menu from './filterButtonGroup'
import ResultTable from './resultTable'
import { connect } from 'react-redux';

class Home extends Component {
  render() {
    return (
      <div className="Home">
        <Grid>
          <Col md={6} >
            <Menu seeAllCheeses={this.props.seeAllCheeses}
                  getRandomCheese={this.props.getRandomCheese}
                  seeSubstitutes={this.props.seeSubstitutes} />
          </Col>
          <Col md={6} >
            <ResultTable cheeses={this.props.results} />
          </Col>
        </Grid>
      </div>
    );
  }
}

const mapStateToPropsHome = (state, ownProps) => {
  return {
    results: state.results,
    query: state.query
  }
}

const mapDispatchToPropsHome = (dispatch, ownProps) => {
  return {
    seeAllCheeses: () => {
      dispatch({type: 'SEE_ALL_CHEESES',
                payload: fetch('http://cheeswhiz.herokuapp.com/api/cheese')
                           .then(function(res) { return res.json(); })
               })
    },
    getRandomCheese: () => {
      dispatch({type: 'GET_RANDOM_CHEESE',
                payload: fetch('http://cheeswhiz.herokuapp.com/api/cheese/random/5/hard')
                           .then(function(res) { return res.json(); })
              })
    },
    onSearchChange: (event) => {
      dispatch({type: 'CHANGE_QUERY',
                newQuery: event.target.value
              })
    },
    seeSubstitutes: () => {
      dispatch({type: 'SEE_SUBSTITUTES',
                payload: fetch('http://cheeswhiz.herokuapp.com/api/cheese/substitute/Brie')
                           .then(function(res) { return res.json(); })
              })
    }
  }
}

const ConnectedHome = connect(mapStateToPropsHome, mapDispatchToPropsHome)(Home);
export default ConnectedHome
