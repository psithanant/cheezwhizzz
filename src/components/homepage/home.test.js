import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Home } from './home';
import toJson from 'enzyme-to-json';

const TEST_CHEESE_LIST = [
  { name: "brie", firmness: "soft", animal: "cow", favorite: 0, id: 1 }
];

describe('Home', () => {
  it('renders', () => {
    const div = document.createElement('div');
    shallow(<Home results={TEST_CHEESE_LIST} />, div);
  });

  test('snapshots', () => {
    const component = shallow(
      <Home results={TEST_CHEESE_LIST} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
