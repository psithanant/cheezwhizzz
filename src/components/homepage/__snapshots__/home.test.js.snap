// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshots 1`] = `
<div
  className="Home"
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
    >
      <Connect(Menu) />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
    >
      <ResultTable
        cheeses={
          Array [
            Object {
              "animal": "cow",
              "favorite": 0,
              "firmness": "soft",
              "id": 1,
              "name": "brie",
            },
          ]
        }
      />
    </Col>
  </Grid>
</div>
`;
